stages/test/test_cloud-init.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/python3
+
+import os.path
+
+from osbuild.testutil.imports import import_module_from_path
+
+
+# Test that the 'datasource_list' option is dumped as a single line
+# in the configuration file.
+def test_cloud_init_datasource_list_single_line(tmp_path):
+    treedir = os.path.join(tmp_path, "tree")
+    config_dir = f"{treedir}/etc/cloud/cloud.cfg.d/"
+    os.makedirs(config_dir, exist_ok=True)
+
+    stage_path = os.path.join(os.path.dirname(__file__), "../org.osbuild.cloud-init")
+    stage = import_module_from_path("stage", stage_path)
+
+    config_filename = "datasource_list.cfg"
+    options = {
+        "filename": config_filename,
+        "config": {
+            "datasource_list": [
+                "Azure",
+            ]
+        }
+    }
+
+    stage.main(treedir, options)
+
+    confpath = f"{treedir}/etc/cloud/cloud.cfg.d/{config_filename}"
```

Comment:
(super nitpick) we could define confpath a bit earlier and safe a few lines as it contains both config_dir and config_filename already. With the new magic of pathlib it could look something like:
```python
    treedir = tmp_path / "tree"
    confpath = treedir / "etc/cloud/cloud.cfg.d/datasource_list.cfg"
    confpath.parent.mkdir(parents=True, exist_ok=True)

    options = {
        "filename": confpath.name,
        "config": {
            "datasource_list": [
                "Azure",
            ]
        }
    }

    stage.main(treedir, options)

    assert confpath.exists()
    assert confpath.read_bytes() == b"datasource_list: [Azure]\n"
```

Reply:
done